lnd.go

```diff
@@ -1041,16 +1047,27 @@ func waitForWalletPassword(cfg *Config, restEndpoints []net.Addr,
 	// Set up a new PasswordService, which will listen for passwords
 	// provided over RPC.
 	grpcServer := grpc.NewServer(serverOpts...)
-	defer grpcServer.GracefulStop()
 	lnrpc.RegisterWalletUnlockerServer(grpcServer, pwService)
 
+	shutdownFuncs := make([]func(), 0)
+	shutdown := func() {
+		// First wait for the wallet unlocker to finish serving its
+		// requests, only then shut down the rest.
+		pwService.Stop()
```

Comment:
this doesn't compile.

Also not needed? If we rely on graceful stop waiting for all requests to be served.